e2etests/test/pnp_telemetry.js

```diff
@@ -0,0 +1,178 @@
+    this.timeout(120000);
+
+    let provisionedDevice, deviceClient, ehClient;
+
+    before(function (beforeCallback) {
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+    });
+
+    afterEach(function (done) {
+      closeDeviceEventHubClients(deviceClient, ehClient, done);
+    });
+
+    it('sends a telemetry message with a component name and the service sees the message with the correct component name, content type, and content encoding', function (done) {
+      const rdv =  new Rendezvous(done);
+      const startAfterTime = Date.now() - 5000;
+
+      const onEventHubMessage = function (eventData) {
+        if (eventData.annotations['iothub-connection-device-id'] === provisionedDevice.deviceId) {
+          if (eventData.properties.content_type !== 'application/json') {
+            debug(`eventData.properties.content_type "${eventData.properties.content_type}" doesn't match "application/json"`);
+          } else if (eventData.properties.content_encoding !== 'utf-8') {
+            debug(`eventData.properties.content_encoding "${eventData.properties.content_encoding}" doesn't match "utf-8"`);
+          } else if (eventData.annotations['dt-subject'] !== componentName) {
+            debug(`eventData.annotations['dt-subject'] "${eventData.annotations['dt-subject']}" doesn't match ${componentName}`);
+          } else if (JSON.stringify(eventData.body) !== JSON.stringify({fake: 'payload'})) {
+            debug(`eventData.body "${JSON.stringify(eventData.body)}" doesn't match "${JSON.stringify({fake: 'payload'})}"`);
+          } else {
+            debug('trying to finish from the receiving side');
+            rdv.imDone('ehClient');
+          }
+        } else {
+          debug(`Incoming device id is: ${eventData.annotations['iothub-connection-device-id']}`);
+        }
+      };
+
+      const onEventHubError = function (err) {
+        debug(`Error from Event Hub Client Receiver: ${err}`);
+        done(err);
+      };
+
+      EventHubClient.createFromIotHubConnectionString(hubConnectionString).then(function (client) {
+        ehClient = client;
+        rdv.imIn('ehClient');
+        return ehClient.getPartitionIds();
+      }).then(function (partitionIds) {
+        partitionIds.forEach(function (partitionId) {
+          ehClient.receive(partitionId, onEventHubMessage, onEventHubError, { eventPosition: EventPosition.fromEnqueuedTime(startAfterTime) });
+        });
+        return new Promise(function (resolve) {
+          setTimeout(resolve, 3000);
+        });
+      }).then(function () {
```

Comment:
Another suggestion:  Every time I see code that iterates over eventhub partitions, I think that this code needs to be refactored.  How often do we do this in E2E tests?  I'm guessing a lot.

Reply:
We do it a lot. There's actually already the `EventHubReceiverHelper` written that I can use that would make things a lot simpler. I'll refactor my tests.

Reply:
Done